common/src/main/java/com/wynntils/functions/InventoryFunctions.java

```diff
@@ -126,18 +126,18 @@ public List<String> getAliases() {
         }
     }
 
-    public static class HeldItemCurrentDurabilityFunction extends Function<Integer> {
+    public static class HeldItemCurrentDurabilityFunction extends Function<String> {
         @Override
-        public Integer getValue(String argument) {
+        public String getValue(String argument) {
             ItemStack itemStack = McUtils.player().getItemInHand(InteractionHand.MAIN_HAND);
 
             Optional<WynnItem> wynnItem = Models.Item.getWynnItem(itemStack);
 
-            if (wynnItem.isPresent() && wynnItem.get() instanceof CraftedGearItem craftedGearItem) {
-                return craftedGearItem.getDurability().getCurrent();
+            if (wynnItem.isPresent() && wynnItem.get() instanceof DurabilityItemProperty durableItem) {
+                return String.valueOf(durableItem.getDurability().getCurrent());
             }
 
-            return -1;
```

Comment:
It feels a bit sad that we should need to sacrifice the type information just to be able to show "no value" in a better way.
What do you say about functions being able to return `Optional<Integer>`, and if the optional is empty, we display nothing?

Reply:
I think this is a task when we finish the info variable refactors of artemis.